l10n_it_split_payment/models/account.py

```diff
@@ -3,49 +3,46 @@
 # Copyright 2016  Alessio Gerace - Agile Business Group
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import _, api, fields, models
+from odoo import _, fields, models
 from odoo.exceptions import UserError
 
-import odoo.addons.decimal_precision as dp
-
 
 class AccountFiscalPosition(models.Model):
     _inherit = "account.fiscal.position"
 
     split_payment = fields.Boolean("Split Payment")
 
 
-class AccountInvoice(models.Model):
-    _inherit = "account.invoice"
+class AccountMove(models.Model):
+    _inherit = "account.move"
 
     amount_sp = fields.Float(
         string="Split Payment",
-        digits=dp.get_precision("Account"),
+        digits="Account",
         store=True,
         readonly=True,
         compute="_compute_amount",
     )
     split_payment = fields.Boolean(
-        "Is Split Payment", related="fiscal_position_id.split_payment"
+        string="Is Split Payment", related="fiscal_position_id.split_payment"
     )
 
-    @api.one
-    @api.depends(
-        "invoice_line_ids.price_subtotal",
-        "tax_line_ids.amount",
-        "tax_line_ids.amount_rounding",
-        "currency_id",
-        "company_id",
-        "date_invoice",
-        "type",
-    )
     def _compute_amount(self):
-        super(AccountInvoice, self)._compute_amount()
-        self.amount_sp = 0
-        if self.fiscal_position_id.split_payment:
-            self.amount_sp = self.amount_tax
-            self.amount_tax = 0
-        self.amount_total = self.amount_untaxed + self.amount_tax
+        super(AccountMove, self)._compute_amount()
```

Comment:
in Python 3 il super lo puoi chiamare anche senza argomenti!

Reply:
eliminato argomenti